include/mqtt/broker/offline_message.hpp

```diff
@@ -96,25 +96,85 @@ class offline_messages {
         catch (packet_id_exhausted_error const& e) {
             MQTT_LOG("mqtt_broker", warning)
                 << MQTT_ADD_VALUE(address, &ep)
-                << e.what();
+                << e.what()
+                << " : send the rest messages later.";
         }
         for (auto const& oflm : messages_) {
             oflm.send(ep);
         }
     }
 
+    void send_by_packet_id_release(endpoint_t& ep) {
+        try {
+            auto& idx = messages_.get<tag_seq>();
+            bool finish = false;
+            while (!idx.empty() && !finish) {
+                idx.modify(
+                    idx.begin(),
+                    [&](auto& e) {
+                        auto qos_value = e.pubopts_.get_qos();
+                        e.send(ep);
```

Comment:
Send is const method, so you don't need idx.modify ?

Reply:
ok.